app/Support/Routing/AjaxRoute.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Plugin\Support\Routing;
+
+use Plugin\Support\Http\Request;
+
+class AjaxRoute extends Route
+{
+    /**
+     * Dispatch the request to the route
+     *
+     * @param \Plugin\Support\Http\Request $request
+     * @return void
+     */
+    public function dispatch(Request $request)
+    {
+        if (in_array('AUTH', $this->methods())) {
+            $this->router->events->listen("wp_ajax_{$this->uri}", $this->handle($request));
+        }
+        if (in_array('GUEST', $this->methods())) {
+            $this->router->events->listen("wp_ajax_nopriv_{$this->uri}", $this->handle($request));
+        }
+    }
```

Comment:
`$this->router->events` should be a protected property on the router instance. Think of another way to access the event dispatcher